uniffi/src/interface/mod.rs

```diff
@@ -699,10 +742,50 @@ impl APIConverter<Method> for weedle::interface::OperationInterfaceMember<'_> {
                 weedle::types::ReturnType::Type(t) => Some(t.resolve_type_definition(ci)?),
             },
             ffi_func: Default::default(),
+            throws: match &self.attributes {
+                Some(attr) => attr.body.list.find_throws(ci)?,
+                None => None,
+            },
         })
     }
 }
 
+trait FindThrows {
```

Comment:
This shouldn't be a trait I think, I'll replace it with a function. I don't think they'll be a difference between the `Throws` attributes on methods and functions

Reply:
Thinking about the way the rest of this code is structured, and about a few of my previous attempts at processing the WebIDL here, here's a hand-wavy suggestion for dealing with attributes. Feel free to ignore it since I really haven't tried it out, I'm just kind of squinting at the problem and thinking about how I might approach it, so the below might turn out to be garbage. But, a suggestion:

* Create a new `enum Attribute` that is specifically for parsing function attributes, with one variant per attribute of interest. Right now I guess you'd only have a single variant `Attribute::Throws(name)`.
* Use `impl TryFrom` to destructure weedle's attribute syntax nodes into your much simpler `Attribute` enum, similar to what we do for other things using the `APIConverter` trait.
    * You don't need a reference to the `ComponentInterface` to achieve this, which is why you can `impl TryFrom` rather than using `APIConverter`.
    * This is where you error out if you encounter an unknown attribute or any of the many syntax forms that weedle supports and we don't.
* In the `APIConverter` impl for `Function`, `Method`, etc where you want to handle attributes:
    * map them into a `Vec` of your `Attribute` enum, like:
        * `let attrs: Vec<Attribute> = attrs.body.list.map(|a| a.try_into()).collect::<Result<_>>()?`
    * inspect the list of converted `Attribute`s to figure out what flags you want to set on your object.

That's a lot of machinery! You'd be justified to conclude that you don't really need all that, and you probably don't *need* it for this single attribute. But the reason I suggested it is to set yourself up for future maintainability as we grow additional attributes, because:

* The "bail if we see unexpected attribute syntax" logic can live in a single place, rather than in each location.
* The logic about what attribute sets what flag on each struct can live with the struct itself.

Reply:
(Also I am just a really huge fan of `TryFrom`...)

Reply:
I `TryFrom`ed it 😄 Added an `Attributes` as a wrapper for `Vec<Attribute>`. And I'm storing that now in methods instead of a single "Error". Can be helpful as we start supporting more attributes I guess, let me know your thoughts 😄